templates/common/_base/files/configure-ovs-network.yaml

```diff
@@ -0,0 +1,313 @@
+    #             lacp: active
+    #             mode: balance-slb
+    #             slaves:
+    #               - name: eth1
+    #               - name: eth2
+    #         - name: br-ex
+    #   - name: br-ex
+    #     type: ovs-interface
+    #     mtu: 9000
+    #     mac-address: 52:54:00:E0:64:7D
+    #
+    # This configuration file format is the same as what is used by nmstate. Only configuration options listed
+    # here are supported. The only configuration this file should contain is for configuring br-ex. Other interfaces
+    # are not supported.
+
+    parse_yaml() {
+      local yaml_file=$1
+      local prefix=$2
+      local s
+      local w
+      local fs
+
+      s='[[:space:]]*'
+      w='[a-zA-Z0-9_.-]*'
+      fs="$(echo @|tr @ '\034')"
+
+      (
+        sed -e '/- [^\“]'"[^\']"'.*: /s|\([ ]*\)- \([[:space:]]*\)|\1-\'$'\n''  \1\2|g' |
+
+        sed -ne '/^--/s|--||g; s|\"|\\\"|g; s/[[:space:]]*$//g;' \
+            -e "/#.*[\"\']/!s| #.*||g; /^#/s|#.*||g;" \
+            -e "s|^\($s\)\($w\)$s:$s\"\(.*\)\"$s\$|\1$fs\2$fs\3|p" \
+            -e "s|^\($s\)\($w\)${s}[:-]$s\(.*\)$s\$|\1$fs\2$fs\3|p" |
+
+        awk -F"$fs" '{
+            indent = length($1)/2;
+            if (length($2) == 0) { conj[indent]="+";} else {conj[indent]="";}
+            vname[indent] = $2;
+            for (i in vname) {if (i > indent) {delete vname[i]}}
+                if (length($3) > 0) {
+                    vn=""; for (i=0; i<indent; i++) {vn=(vn)(vname[i])("_")}
+                    printf("%s%s%s%s=(\"%s\")\n", "'"$prefix"'",vn, $2, conj[indent-1],$3);
+                }
+            }' |
+
+        sed -e 's/_=/+=/g' |
+
+        awk 'BEGIN {
+                FS="=";
+                OFS="="
+            }
+            /(-|\.).*=/ {
+                gsub("-|\\.", "_", $1)
+            }
+            { print }'
+      ) < "$yaml_file"
+    }
+
+    unset_variables() {
+      # Pulls out the variable names and unsets them.
+      local variable_string="$@"
+      unset variables
+      variables=()
+      for variable in ${variable_string[@]}; do
+        variables+=($(echo $variable | grep '=' | sed 's/=.*//' | sed 's/+.*//'))
+      done
+      for variable in ${variables[@]}; do
+        unset $variable
+      done
+    }
+
+    create_variables() {
+      local yaml_file="$1"
+      local prefix="$2"
+      local yaml_string="$(parse_yaml "$yaml_file" "$prefix")"
+      unset_variables ${yaml_string[@]}
+      eval "${yaml_string}"
+    }
+
+    CONFIG_FILE="/etc/sysconfig/network-scripts/ovs-config.yaml"
+
+    iface=""
+    counter=0
+    # find default interface
+    while [ $counter -lt 12 ]; do
+      iface=$(ip -j route show default | jq -r '.[0].dev')
+      if [ -n "$iface" ]; then
+        echo "Default gateway interface found: ${iface}"
+        break
+      fi
+      counter=$((counter+1))
+      echo "No default route found on attempt: ${counter}"
+      sleep 5
+    done
+
+    if [ "$iface" = "br-ex" ]; then
```

Comment:
can the name of the bridge change to anything else other than `br-ex`? Is this something that should be passed in as an argument to ExecStart?

Reply:
It will always be br-ex. We have to pass this name into ovn-k8s via CNO, and if we can stick to a single name it will make debugging easier.